app/controllers/tabbed_controller.rb

```diff
@@ -8,9 +8,11 @@ def show
     @page_title = @item.title
 
     if params[:tab]
-      fail ActiveRecord::RecordNotFound unless @item.tabs[params[:tab]]
+      fail IndexError unless @item.tabs[params[:tab]]
     else
       redirect_to('/' + params[:controller] + '/' + @item.tab_path) if @item.tab_path != params[:id]
     end
   end
+
+  rescue_from IndexError, with: :render_404
```

Comment:
string: It's coming from the routes. I have no strong preferences on what the particular error is... You could argue for making a new class that is only used here, so we can be absolutely sure what we're catching.